python/GafferUITest/PathListingWidgetTest.py

```diff
@@ -1277,6 +1290,53 @@ def testStretchLastColumnIsSetFromColumns( self ) :
 		widget.setColumns( [ GafferUI.PathListingWidget.StandardColumn( "Test", "test" ) ] )
 		self.assertTrue( widget._qtWidget()._shouldStretchLastColumn() )
 
+	def testModelNameSortingDoesntEmitDataChanged( self ) :
+
+		path = self.InfinitePath( "/" )
+
+		widget = GafferUI.PathListingWidget(
+			path = path,
+			displayMode = GafferUI.PathListingWidget.DisplayMode.Tree,
+			sortable = False,
+			columns = [
+				GafferUI.PathListingWidget.defaultNameColumn,
+				GafferUI.PathListingWidget.StandardColumn( "Test", "test" )
+			],
+		)
+
+		expansionDepth = 4
+		model = widget._qtWidget().model()
+
+		self.__expandModel( model, depth = expansionDepth )
+		_GafferUI._pathModelWaitForPendingUpdates( GafferUI._qtAddress( model ) )
+
+		changes = []
+		def dataChanged( *args ) :
+			changes.append( args )
+
+		model.dataChanged.connect( dataChanged )
+
+		# setting our widget to be sortable shouldn't result in any data changes
+		# as we're sorting by the `name` column
+
+		widget.setSortable( True )
+		_GafferUI._pathModelWaitForPendingUpdates( GafferUI._qtAddress( model ) )
+		self.assertEqual( len( changes ), 0 )
+
+		# likewise, changing the sort order of the `name` column shouldn't result
+		# in data changes
+
+		widget._qtWidget().header().setSortIndicator( 0, QtCore.Qt.AscendingOrder )
+		_GafferUI._pathModelWaitForPendingUpdates( GafferUI._qtAddress( model ) )
+		self.assertEqual( len( changes ), 0 )
```

Comment:
Would be nice to assert that the model has actually been sorted into the expected order here.

Reply:
Added testing of expected sort orders (and swapped the second sort to Descending in order to actually see a change 🙃 ) in 2de42a1.